pagopa/io-app
# feat(Piattaforma Notifiche): [IAMVL-81] Let the user disable the disclaimer before opening a message from PN

## Short description
### ⚠️ **This PR depends on #4033** 

This PR adds the implementation for handling the "Don't show me again" toggle in the disclaimer alert when opening a message from PN.

## List of changes proposed in this pull request
- Add a new preferences section for PN in the store
- Add the related action/reducer
- Modify the `useMessageOpening` logic to check for user's preference

## How to test
Use the [IIP-18-add-pn-endpoints](https://github.com/pagopa/io-dev-api-server/tree/IIP-18-add-pn-endpoints) branch of the dev server and add this to your `config.json`:
```
{
  "messages": {
    "pnCount": 1
  },
  "services": {
    "includePn": true
  }
}
```
Set `PN_ENABLED=YES` to `.env` and run the app, then:
1. select a message from PN;
2. select "Don't show me again" in the bottom sheet and open the message;
3. go back;
4. re-select the message.

The message should be opened without any alert. The preferences have effect for all the messages coming from PN (not just the one you first open).
